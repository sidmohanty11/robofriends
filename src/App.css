.App h1 {
	font-size: 40px;
	text-align: center;
	color: lightgreen;
	font-family: 'Major Mono Display', monospace;
}

.App span {
	margin-bottom: 5px;
}