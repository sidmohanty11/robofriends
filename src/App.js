import React, { useState, useEffect } from 'react';
import './App.css';
import Card from './Card';
import SearchBox from './SearchBox';
import Scroll from './Scroll';

function App() {
  const [robots, setRobots] = useState([]);
  const [searchfield, setSearchField] = useState('');

  useEffect(() => { 
    async function addRobots() {
        const res = await fetch("https://jsonplaceholder.typicode.com/users");
        const data = await res.json();
        setRobots(data);
    }
    addRobots();
  }, []);

  const onSearchChange = (e) => {
    setSearchField(e.target.value);
  }
  const filteredRobots = robots.filter((robot) => {
      return robot.name.toLowerCase().includes(searchfield.toLowerCase());
  });

  return (
    <div className="App tc">
    <h1>ROBOFRIENDS!</h1>
    <SearchBox searchfield={searchfield} onSearchChange={onSearchChange} />
    <Scroll>
      {filteredRobots? 
        filteredRobots.map((robot) => (
          <Card 
          key={robot.id} 
          id={robot.id} 
          name={robot.name} 
          username={robot.username}  
          email={robot.email} />
          )): 
          robots.map((robot) => (
          <Card 
          key={robot.id} 
          id={robot.id} 
          name={robot.name} 
          username={robot.username}  
          email={robot.email} />
        ))}
    </Scroll>
    </div>
  );
}

export default App;
