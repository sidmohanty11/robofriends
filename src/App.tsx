import React, { useState, useEffect, ChangeEvent } from 'react';
import './App.css';
import Card from './Card';
import SearchBox from './SearchBox';
import Scroll from './Scroll';

type robot = {
  id: number,
  username: string,
  email: string,
  name: string,
}

function App() {
  const [robots, setRobots] = useState([]);
  const [searchfield, setSearchField] = useState('');

  useEffect(() => { 
    async function addRobots() {
        const res = await fetch("https://jsonplaceholder.typicode.com/users");
        const data = await res.json();
        setRobots(data);
    }
    addRobots();
  }, []);

  const onSearchChange = (e: ChangeEvent<HTMLInputElement>) => {
    setSearchField(e.target.value);
  }
  const filteredRobots = robots.filter((robot: robot): boolean => {
      return robot.name.toLowerCase().includes(searchfield.toLowerCase());
  });

  return (
    <div className="App tc">
    <h1>ROBOFRIENDS!</h1>
    <SearchBox searchfield={searchfield} onSearchChange={onSearchChange} />
    <Scroll>
      {filteredRobots? 
        filteredRobots.map((robot: robot) => (
          <Card 
          key={robot.id} 
          id={robot.id} 
          name={robot.name} 
          username={robot.username}  
          email={robot.email} />
          )): 
          robots.map((robot: robot) => (
          <Card 
          key={robot.id} 
          id={robot.id} 
          name={robot.name} 
          username={robot.username}  
          email={robot.email} />
        ))}
    </Scroll>
    </div>
  );
}

export default App;
