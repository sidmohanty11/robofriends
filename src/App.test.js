import App from './App';
import { shallow } from 'enzyme';

it("check if app component is nicely rendered", () => {
    expect(shallow(<App />)).toMatchSnapshot();
})