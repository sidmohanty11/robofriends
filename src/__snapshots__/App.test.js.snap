// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check if app component is nicely rendered 1`] = `
<div
  className="App tc"
>
  <h1>
    ROBOFRIENDS!
  </h1>
  <SearchBox
    onSearchChange={[Function]}
    searchfield=""
  />
  <Scroll />
</div>
`;
